src/instruction/instructions/mov/movs.ts

```diff
@@ -65,12 +65,14 @@ export class MovsImmediate8Instruction extends BaseInstruction {
   private rdPattern: string = '00100XXX00000000'
   private immPattern: string = '00100000XXXXXXXX'
   private expectedOptionCount: number = 2
+  private instrWithSameName: BaseInstruction[] = [
+    new MovsRegistersInstruction()
+  ]
 
   public canEncodeInstruction(name: string, options: string[]): boolean {
     return (
       super.canEncodeInstruction(name, options) &&
-      isOptionCountValid(options, this.expectedOptionCount) &&
-      isImmediate(options[1])
+      !this.instrWithSameName.some((i) => i.canEncodeInstruction(name, options))
```

Comment:
Isn't it possible to also place this code in the super class BaseIntruction? You could make a new abstract array in the super class that is empty by default and can optionally be overridden by the child class so that the cross checks are being made.

Reply:
made the requested changes but not sure if it makes more readable